src/Operation/BulkWrite.php

```diff
@@ -305,7 +306,7 @@ private function validateOperations(array $operations, ?DocumentCodec $codec, En
                 case self::INSERT_ONE:
                     // $args[0] was already validated above. Since DocumentCodec::encode will always return a Document
                     // instance, there is no need to re-validate the returned value here.
-                    if ($codec) {
+                    if ($codec && is_object($args[0])) {
```

Comment:
Alternatively, we can throw a `MongoDB\Exception\UnsupportedValueException` when the value is not an object.